test/Concurrency/Runtime/async_task_sleep.swift

```diff
@@ -45,4 +46,37 @@ import Dispatch
     // CHECK: Run second
     await task.get()
   }
+
+  static func testSleepForUIntMax_nanoseconds() async {
+    let t = Task.detached {
+      print("Run infinite sleep")
+      try? await Task.sleep(nanoseconds: .max)
+      print("Infinite sleep done (cancelled: \(Task.isCancelled)) @ \(#function)")
+      precondition(Task.isCancelled, "Infinite sleep finished yet task was NOT cancelled! @ \(#function)")
+    }
+
+    // CHECK: Infinite sleep done (cancelled: true)
+    try? await Task.sleep(nanoseconds: 500_000_000) // 500ms
+    t.cancel()
+
+    await t.value
+  }
+
+  // FIXME: can't quite figure this one out, cc Philippe
+  // Uses a different internal API, so we need to test this separately
+  static func testSleepForUIntMax_untilDeadline() async {
+    let t = Task.detached {
+      print("Run infinite sleep")
+      try? await Task.sleep(for: .nanoseconds(UInt64.max))
+      print("Infinite sleep done (cancelled: \(Task.isCancelled)) @ \(#function)")
```

Comment:
I think we should adjust it on the Swift side to clamp the value there; because testing it like this gets a duration of `709551615000000000 seconds, 18446744073 attoseconds`

Reply:
Hm so just cap it at Int64.max on the swift side perhaps...

Reply:
yea, that seems reasonable to me (or cap and loop)